File: dashboard/src/types/bot.ts
import type { Agent } from './agent'

/**
 * 机器人状态枚举
 */
export enum BotStatus {
  Active = 'active',
  Inactive = 'inactive',
}

/**
 * 机器人状态显示名称映射
 */
export const BotStatusLabels: Record<BotStatus, string> = {
  [BotStatus.Active]: '启用',
  [BotStatus.Inactive]: '禁用',
}

/**
 * 机器人状态选项（用于下拉选择）
 */
export const BotStatusOptions = [
  { value: BotStatus.Active, label: BotStatusLabels[BotStatus.Active] },
  { value: BotStatus.Inactive, label: BotStatusLabels[BotStatus.Inactive] },
]

/**
 * 机器人平台类型枚举
 */
export enum BotPlatform {
  QQ = 'qq',
  Feishu = 'feishu',
}

/**
 * 机器人平台显示名称映射
 */
export const BotPlatformLabels: Record<BotPlatform, string> = {
  [BotPlatform.QQ]: 'QQ',
  [BotPlatform.Feishu]: '飞书',
}

/**
 * 机器人平台选项（用于下拉选择）
 */
export const BotPlatformOptions = [
  { value: BotPlatform.QQ, label: BotPlatformLabels[BotPlatform.QQ] },
  { value: BotPlatform.Feishu, label: BotPlatformLabels[BotPlatform.Feishu] },
]

/**
 * 机器人配置
 */
export interface Bot {
  id: string
  name: string
  description: string
  platform: BotPlatform
  identifier: string
  config: Record<string, unknown>
  agent_id: string | null
  agent?: Agent
  status: BotStatus
  created_at: string
  updated_at: string
}

/**
 * 创建机器人请求
 */
export interface CreateBotRequest {
  name: string
  description?: string
  platform: BotPlatform
  identifier?: string
  config?: Record<string, unknown>
  agent_id?: string
}

/**
 * 更新机器人请求
 */
export interface UpdateBotRequest {
  name?: string
  description?: string
  identifier?: string
  config?: Record<string, unknown>
  status?: BotStatus
}

/**
 * 绑定智能体请求
 */
export interface BindAgentRequest {
  agent_id: string | null
}

/**
 * QQ 机器人配置
 */
export interface QQBotConfig {
  app_id: string
  app_secret: string
}

/**
 * 飞书机器人配置
 */
export interface FeishuBotConfig {
  app_id: string
  app_secret: string
  encrypt_key?: string
}

File: beepbot/internal/types/bot.go
package types

import (
	"time"

	"gorm.io/datatypes"
)

// BotStatus 机器人状态
type BotStatus string

const (
	BotStatusActive   BotStatus = "active"
	BotStatusInactive BotStatus = "inactive"
)

// BotPlatform 机器人平台类型
type BotPlatform string

const (
	BotPlatformQQ     BotPlatform = "qq"
	BotPlatformFeishu BotPlatform = "feishu"
	// 未来可扩展: discord, telegram, wechat 等
)

// Bot IM机器人配置
type Bot struct {
	ID          string         `json:"id" gorm:"column:id;primaryKey;type:varchar(64)"`
	Name        string         `json:"name" gorm:"column:name;type:varchar(128);not null;uniqueIndex"`
	Description string         `json:"description" gorm:"column:description;type:text"`
	Platform    BotPlatform    `json:"platform" gorm:"column:platform;type:varchar(32);not null;index"`
	Identifier  string         `json:"identifier" gorm:"column:identifier;type:varchar(128)"`
	Config      datatypes.JSON `json:"config" gorm:"column:config;type:jsonb"`
	AgentID     *string        `json:"agent_id" gorm:"column:agent_id;type:varchar(64);index"`
	Status      BotStatus      `json:"status" gorm:"column:status;type:varchar(16);default:active;index"`
	CreatedAt   time.Time      `json:"created_at" gorm:"column:created_at;type:timestamptz;not null"`
	UpdatedAt   time.Time      `json:"updated_at" gorm:"column:updated_at;type:timestamptz;not null"`

	// 关联
	Agent *Agent `json:"agent,omitempty" gorm:"foreignKey:AgentID"`
}

// BotQuery 机器人查询参数
type BotQuery struct {
	Name     string      // 名称模糊搜索
	Status   BotStatus   // 状态筛选
	Platform BotPlatform // 平台筛选
}

// TableName 指定表名
func (Bot) TableName() string {
	return "bots"
}
